### PR TITLE
Fix histogram snapshotting after a lapse in snapshotting     

### PR DESCRIPTION
**Pull request by olix0r**
Problem

When snapshotting does not occur for serveral minutes, the 60-second snapshotting is not honored. If snapshotting does not occur for two minutes, histogram windowin will not be properly honored until snapshot has been called twice.

Solution

Change snapshot windowing to be based on the current time.
 
Fixes twitter/twitter-server#33

### CODE REVIEW

**Comment by olix0r** [comment]
@yokoshin i put together a proof of concept test for your change.  Please review and let me know if it looks right.

**Comment by kevinoliver** [comment]
lgtm @olix0r

**Comment by yokoshin** [comment]
lgtm @olix0r

File: finagle-stats/src/test/scala/com/twitter/finagle/stats/MetricsBucketedHistogramTest.scala [hunk 1]
```diff
@@ -132,15 +132,63 @@ class MetricsBucketedHistogramTest extends FunSuite {
...
 
       h.add(Int.MaxValue)
       // roll to window 2 (this should make data A visible after a call to snapshot)
       roll()
-      val countsSnap1 = details.counts
-      assert(countsSnap1 == Seq(BucketAndCount(1, 2, 1),
+      assert(details.counts == Seq(BucketAndCount(1, 2, 1),
         BucketAndCount(2137204091, Int.MaxValue, 1)))
+
+      // Advance several minutes without snapshotting (i.e. to
```
Comment (mosesn) [inline finagle-stats/src/test/scala/com/twitter/finagle/stats/MetricsBucketedHistogramTest.scala]: break this off into its own test?

**Comment by mosesn** [comment]
LGTM except for a nit in the test

File: finagle-stats/src/test/scala/com/twitter/finagle/stats/MetricsBucketedHistogramTest.scala (hunk 1 above)
Comment (olix0r) [inline finagle-stats/src/test/scala/com/twitter/finagle/stats/MetricsBucketedHistogramTest.scala]: The test name, _histogram snapshot respects refresh window_, seemed appropriate...  i'll split it out if that's clearer, though.

Comment (mosesn) [inline finagle-stats/src/test/scala/com/twitter/finagle/stats/MetricsBucketedHistogramTest.scala]: I don't mind the test name, but I don't like tests that are too long.  it makes it hard to tell what has broken when a test fails, and it's harder to tell what a test is trying to do when you read the code.

**Comment by olix0r** [comment]
updated test

**Comment by mosesn** [comment]
LGTM

**Comment by cacoco** [comment]
This has been merged internally and should be pushed on the next sync.

**Comment by kevinoliver** [comment]
looks like this landed in 186f7a42f but we messed up and lost the commit info :( 
apologies.

**Pull request closed without merging**